### ISSUE TITLE
Declarative limits

### ISSUE THREAD

**Issue by jezzsantos** [wwwlicious/servicestack-ratelimit-redis#2]
Hi Scott,

I am very tempted to use this plugin, but what is stopping me is that Id much rather declaratively set my limits on each operation/dto as an attribute, instead of in configuration settings.

Something like this in a service:

```
[LimitRate(5, 1)]
[LimitRate(15, 60)]
[LimitRate(100, 3600)]
public MyDtoReponse Post (MyDto request)
{
...
}

(Using the same rate limit set you demonstrate in the readme)

Is there some way to extend what you have to support this?

**Comment by wwwlicious** [comment]
Hi @jezzsantos 

This would require a new implementation of the  `ILimitProvider` that read the LimitRateAttribute values from the request DTO type, probably initialized on creation using reflection type scanning.

The plugin LimitProvider property could then just be set to use it.

Would welcome a PR to add this functionality if you decide to try this route.

**Comment by jezzsantos** [comment]
Hey Scott, thanks.
I might give that a go, if you are saying conceptually it would work given the framework you have around it?
Anything you can see that might be a challenge for me?

**Comment by jezzsantos** [comment]
Would you accept a version boost of ServiceStack to at least 4.5.8? or perhaps 4.5.14 (last version before 5.0.0)?

**Comment by wwwlicious** [comment]
@jezzsantos  latest v4.5 would be fine. 👍 
I think conceptually it will work.

**Comment by jezzsantos** [comment]
Hey @wwwlicious 

Well I started work on this, but I'm afraid I wont be able to go to 4.5.14, because it contains a bug that is not fixed until 5.0.2. (see https://forums.servicestack.net/t/stream-endpoint-change-after-4-5-8/6185/4).

My choices with my PR for this repo are go from 4.5.8 to 4.5.12, OR 4.5.8 to 5.0.2. (or 5.1.0).
Obviously the latter forces a massive piece of work on me (migrating from SS 4.X to 5.X)  to move our other codebase before I can fully integrate this plugin.

**Comment by wwwlicious** [comment]
@jezzsantos any option to merge the fix back into the 4.x branch?

**Comment by jezzsantos** [comment]
Yep, just got a workaround: https://forums.servicestack.net/t/stream-endpoint-change-after-4-5-8/6185/6.
I doubt Demis will release a 4.X version with this fix.

I am at least happy to go to 4.5.14 now! 👍

**Comment by jezzsantos** [comment]
Hey @wwwlicious 

Having a crap time trying to get a PR to test properly, can you have a look at #3? 

The license is expired on the repo, and parallel testing cannot work since each test class will try to assign the `ServiceStackHost` to the same AppDomain.

**Comment by jezzsantos** [comment]
Sorry mate, gave up waiting for you to respond. Went our own way.

**Comment by wwwlicious** [comment]
no problem
